tests/ignite/metrics/nlp/test_bleu.py

```diff
@@ -28,117 +29,185 @@ def test_wrong_inputs():
         Bleu().compute()
 
+    with pytest.raises(ValueError, match='Average must be either "macro" or "micro"'):
+        Bleu(average="macros")
 
-@pytest.mark.parametrize(
-    "candidate, references",
+
+parametrize_args = (
+    "candidates, references",
     [
-        ([["a"], ["a"]]),
         ([["a", "a", "a", "b", "c"]], [[["a", "b", "c"], ["a", "a", "d"]]]),
         corpus.sample_1,
         corpus.sample_2,
         corpus.sample_3,
         corpus.sample_4,
     ],
 )
-def test_corpus_bleu(candidate, references):
-    print(candidate, references)
+
+
+def _test(candidates, references, average):
     for i in range(1, 8):
         weights = tuple([1 / i] * i)
-        with warnings.catch_warnings():
-            warnings.simplefilter("ignore")
-            reference = corpus_bleu(references, candidate, weights=weights)
-        bleu = Bleu(ngram=i)
-        assert pytest.approx(reference) == bleu._corpus_bleu(references, candidate)
-        bleu.update((candidate, references))
+        bleu = Bleu(ngram=i, average=average)
+
+        if average == "macro":
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore")
+                reference = sentence_bleu(references[0], candidates[0], weights=weights)
+            assert pytest.approx(reference) == bleu._sentence_bleu(references[0], candidates[0])
+
+        elif average == "micro":
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore")
+                reference = corpus_bleu(references, candidates, weights=weights)
+            assert pytest.approx(reference) == bleu._corpus_bleu(references, candidates)
+
+        bleu.update((candidates, references))
         assert pytest.approx(reference) == bleu.compute()
 
 
-@pytest.mark.parametrize(
-    "candidate, references",
-    [
-        ([["a", "a", "a", "b", "c"]], [[["a", "b", "c"], ["a", "a", "d"]]]),
-        corpus.sample_1,
-        corpus.sample_2,
-        corpus.sample_3,
-        corpus.sample_4,
-    ],
-)
-def test_corpus_bleu_smooth1(candidate, references):
+@pytest.mark.parametrize(*parametrize_args)
+def test_macro_bleu(candidates, references):
+    _test(candidates, references, "macro")
+
+
+@pytest.mark.parametrize(*parametrize_args)
+def test_micro_bleu(candidates, references):
+    _test(candidates, references, "micro")
+
+
+def _test_smooth(candidates, references, average):
     for i in range(1, 8):
         weights = tuple([1 / i] * i)
-        with warnings.catch_warnings():
-            warnings.simplefilter("ignore")
-            reference = corpus_bleu(
-                references, candidate, weights=weights, smoothing_function=SmoothingFunction().method1
-            )
-        bleu = Bleu(ngram=i, smooth="smooth1")
-        assert reference == bleu._corpus_bleu(references, candidate)
-        bleu.update((candidate, references))
+        bleu = Bleu(ngram=i, smooth="smooth1", average=average)
```

Comment:
This method could be merged with the previous test one. You should add a parameter `smooth` for the bleu metric with default value `no_smooth` and a parameter `smoother_nltk_fn` with None default value. Thus, only one test method 

```python
def _test(candidate, references, average, smooth="no_smooth", smooth_nltk_fn=None):
    for i in range(1, 8):
        weights = tuple([1 / i] * i)
        with warnings.catch_warnings():
            warnings.simplefilter("ignore")
            nltk_args = (references[0], candidate[0], weights=weights, smoothing_function=smooth_nltk_fn)
            if average == "macro":
                reference = sentence_bleu(*nltk_args)
            else:
                reference = corpus_bleu(*nltk_args)
        bleu = Bleu(ngram=i, average=average, smooth=smooth)
        if average == "macro":
            assert pytest.approx(reference) == bleu._sentence_bleu(references[0], candidate[0])
        else:
            assert pytest.approx(reference) == bleu._corpus_bleu(references[0], candidate[0])
        bleu.update((candidate, references))
        assert pytest.approx(reference) == bleu.compute()

@pytest.mark.parametrize(*parametrize_arguments)
def test_macro_bleu(candidate, references):
    _test(candidate, references, average="macro")

@pytest.mark.parametrize(*parametrize_arguments)
def test_micro_bleu(candidate, references):
    _test(candidate, references, average="micro")

@pytest.mark.parametrize(*parametrize_args)
 def test_macro_bleu_smooth1(candidates, references):
     _test_smooth(candidates, references, "macro", "smooth1", SmoothingFunction().method1)

@pytest.mark.parametrize(*parametrize_args)
 def test_micro_bleu_smooth1(candidates, references):
     _test_smooth(candidates, references, "micro", "smooth1", SmoothingFunction().method1) 
```